Sources/Client/Utils/Atomic.swift

```diff
@@ -13,12 +13,25 @@ import Foundation
 public final class Atomic<T> {
     /// A didSet callback type.
     public typealias DidSetCallback = (_ value: T?, _ oldValue: T?) -> Void
-    
-    private let queue = DispatchQueue(label: "io.getstream.Chat.Atomic", qos: .userInitiated, attributes: .concurrent)
-    private var value: T?
+
+    private let lock = NSRecursiveLock()
+
+    private var value: T? {
```

Comment:
Can we now avoid an optional type?

Reply:
Yes, it's possible but I didn't want to do it in this PR.